### PR TITLE
[chore] Change receiver config tests to unmarshal config only for that component. (part3)

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Change receivers config tests to unmarshal config only for that component.

- [x] jaegerreceiver
- [x] memcachedreceiver
- [x] mongodbatlasreceiver
- [x] mongodbreceiver
- [x] mysqlreceiver
- [x] nginxreceiver
- [x] nsxtreceiver
- [x] opencensusreceiver
- [x] otlpjsonfilereceiver
- [x] podmanreceiver
- [x] postgresqlreceiver

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13224

### CODE REVIEW

File: receiver/mongodbreceiver/testdata/config.yaml [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-exporters:
-  nop:
-
-service:
-  pipelines:
-    metrics:
-      receivers: [mongodb]
-      processors: [nop]
-      exporters: [nop]
+mongodb:
```
Comment (djaglowski) [inline receiver/mongodbreceiver/testdata/config.yaml:1]: I've noted in #14567 that this and several other receivers I'm responsible for are missing a test that loads and validates this config. For the sake of avoiding merge conflicts, I'll add these tests after this set of PRs is merged.

**Review by djaglowski — APPROVED** [review]

File: receiver/jaegerreceiver/trace_receiver_test.go [hunk 2]
```diff
@@ -286,10 +286,9 @@ func expectedTraceData(t1, t2, t3 time.Time) ptrace.Traces {
 	return traces
 }
 
-func grpcFixture(t *testing.T, t1 time.Time, d1, d2 time.Duration) *api_v2.PostSpansRequest {
+func grpcFixture(t1 time.Time, d1, d2 time.Duration) *api_v2.PostSpansRequest {
 	traceID := model.TraceID{}
-	require.NoError(t, traceID.Unmarshal([]byte{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80}))
-
+	traceID.Unmarshal([]byte{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8, 0xF9, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x80}) // nolint:errcheck
```
Comment (mx-psi) [inline receiver/jaegerreceiver/trace_receiver_test.go:291]: This PR seems to have reverted #14540, was it intentional? If not, would you mind to open a PR to bring back these changes @kovrus ?

Comment (kovrus) [reply receiver/jaegerreceiver/trace_receiver_test.go:291]: No, it was not an intentional change. I'll bring it back.

Comment (mx-psi) [reply receiver/jaegerreceiver/trace_receiver_test.go:291]: Awesome, thanks :) Feel free to ping me on the PR if you want a review

**Pull request merged**